search/client.go

```diff
@@ -110,3 +109,25 @@ func (c *client) ListFacetValues(field string, maximumNumberOfValues int) (*Face
 	err = json.NewDecoder(resp.Body).Decode(facetValues)
 	return facetValues, nil
 }
+
+func createRequest(client *client, reqType string, queryStringParams map[string]string, customHeaders map[string]string, endpoint string, body io.Reader) (*http.Request, error) {
```

Comment:
I don't really see the need to create a function here especially since it requires you to create all those duplicates data structures.

If you really want to, you could create a new object that represents the common headers that are always present on the request and add a decorator to the request to `addHeaders(req, new ReqHeaders{...})` otherwise, leave it where it was with the `req.Header.Add`